signer/src/block_observer.rs

```diff
@@ -416,47 +417,110 @@ impl<C: Context, B> BlockObserver<C, B> {
         &self,
         block_hash: BlockHash,
         txs: &[BitcoinTxInfo],
+    ) -> Result<(), Error> {
+        // The first time, we get all sweep transactions with inputs that
+        // we know about. However, we could have locked the UTXO with a new
+        // scriptPubKey, and we have no way of knowing that ahead of time.
+        // The first pass over will populate the database with the new
+        // scriptPubKeys.
+        self.extract_sbtc_transactions_inner(block_hash, txs)
+            .await?;
+        // This will catch cases where the signers have locked up their
+        // UTXO with a new scriptPubKey and there are a chain of
+        // transactions in the block.
+        self.extract_sbtc_transactions_inner(block_hash, txs).await
```

Comment:
sanity check: we need the second pass only because the chain can be "out of order" inside the block? If so, then isn't the `signer_script_pubkeys.insert` in `extract_sbtc_transactions_inner` redundant? Given we always call that function twice, andwe can at most one change of aggregate key in a block.

Reply:
Updated in https://github.com/stacks-sbtc/sbtc/pull/1669/commits/00cf64a34d76892aa2e1cc6823fab5c3cd077d3d